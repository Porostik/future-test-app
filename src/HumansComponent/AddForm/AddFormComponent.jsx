import React from 'react';
import { Formik, Form, Field } from 'formik';

import { displayedProps, buttonsText, inputPlaceholders } from 'constants.js';
import { Input, Button } from 'shared';
import { validation } from './validators';
import { WhithError } from 'hoc';

// Презентационный компонент формы добавления чевека
// initObj - начальный значение полей формы
function AddFormComponent({ initObj, handleSubmit }) {
  const FromInput = WhithError(Input);

  return (
    <Formik
      initialValues={initObj}
      validateOnChange={false}
      validateOnBlur={true}
      validationSchema={validation}
      onSubmit={(values) => {
        handleSubmit({ ...values });
      }}>
      {({ submitForm, errors, touched }) => (
        <Form className="add-form">
          {displayedProps.map((item, index) => (
            <Field
              key={`${item}_${index}`}
              name={item}
              as={FromInput}
              placeholder={inputPlaceholders[item]}
              errorMessage={errors[item]}
              isToched={touched[item]}
            />
          ))}
          <Button onClick={submitForm} text={buttonsText.addForm} />
        </Form>
      )}
    </Formik>
  );
}

export default AddFormComponent;
