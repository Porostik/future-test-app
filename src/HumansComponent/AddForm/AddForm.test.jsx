import React from 'react';
import { shallow, mount } from 'enzyme';
import AddFormComponent from './AddFormComponent';

describe('AddForm component', () => {
  test('AddForm snashot', () => {
    const initObj = {
      id: '',
      firstName: '',
      lastName: '',
      email: '',
      phone: '',
    };
    const wrapper = shallow(<AddFormComponent handleSubmit={jest.fn()} initObj={initObj} />);
    expect(wrapper).toMatchSnapshot();
  });
});
