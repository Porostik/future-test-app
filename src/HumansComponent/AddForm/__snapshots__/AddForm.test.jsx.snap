// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddForm component AddForm snashot 1`] = `
<Formik
  initialValues={
    Object {
      "email": "",
      "firstName": "",
      "id": "",
      "lastName": "",
      "phone": "",
    }
  }
  onSubmit={[Function]}
  validateOnBlur={true}
  validateOnChange={false}
  validationSchema={
    ObjectSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_excludedEdges": Array [],
      "_nodes": Array [
        "phone",
        "email",
        "lastName",
        "firstName",
        "id",
      ],
      "_sortErrors": [Function],
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {},
      "fields": Object {
        "email": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "email": false,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "label": undefined,
            "meta": undefined,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "firstName": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "matches": false,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "label": undefined,
            "meta": undefined,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "id": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "matches": false,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "label": undefined,
            "meta": undefined,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "lastName": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "matches": false,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "label": undefined,
            "meta": undefined,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "phone": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "matches": false,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "label": undefined,
            "meta": undefined,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
      },
      "spec": Object {
        "abortEarly": true,
        "label": undefined,
        "meta": undefined,
        "nullable": false,
        "presence": "optional",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
      "type": "object",
    }
  }
>
  <Component />
</Formik>
`;
